temporalio/exceptions.py

```diff
@@ -65,10 +68,14 @@ def __init__(
         message: str,
         *details: Any,
         type: Optional[str] = None,
-        non_retryable: bool = False
+        non_retryable: bool = False,
     ) -> None:
         """Initialize an application error."""
-        super().__init__(message)
+        super().__init__(
+            message,
+            # If there is a type, prepend it to the message on the string repr
+            exc_args=(message if not type else f"{type}: {message}",),
```

Comment:
Why not override `__str__`?

Reply:
I want to alter the `Exception.args` not just the string. That way our exception, like basically all other Python exceptions, still has `str(exception) == str(exception.args)`. Doesn't seem to have benefit only alter the string representation.